src/loader.js

```diff
@@ -284,6 +284,48 @@ function setupModuleLoader(window) {
 
           /**
            * @ngdoc method
+           * @name angular.Module#component
+           * @module ng
+           * @param {string} name Component name
+           * @param {Object} options Component definition object
+           * @description
+           * TODO...
+           */
+          component: function(name, options) {
```

Comment:
this is just a wrapper around module.directive. module.directive will do the invokeLaterAndSetModuleName